### ISSUE TITLE
Detaching from a remote program throws an error

### ISSUE THREAD

**Issue by DonJayamanne** [Microsoft/ptvsd#317]
When detaching from an attached (remote debugging) process throws the following error:
```
Exception in thread ptvsd.Server:
Traceback (most recent call last):
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/ipcjson.py", line 292, in process_one_message
    msg = self.__message.pop(0)
IndexError: pop from empty list

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/donjayamanne/anaconda3/lib/python3.6/threading.py", line 916, in _bootstrap_inner
    self.run()
  File "/Users/donjayamanne/anaconda3/lib/python3.6/threading.py", line 864, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/ipcjson.py", line 286, in process_messages
    if self.process_one_message():
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/ipcjson.py", line 295, in process_one_message
    self._wait_for_message()
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/ipcjson.py", line 161, in _wait_for_message
    line = self._buffered_read_line_as_ascii()
  File "/Users/donjayamanne/Desktop/Development/vscode/ptvsd/ptvsd/ipcjson.py", line 120, in _buffered_read_line_as_ascii
    temp = self.__socket.recv(1024)
ConnectionResetError: [Errno 54] Connection reset by peer

```

**Comment by DonJayamanne** [comment]
@ericsnowcurrently I'm assuming you'll be fixing this, either way, I'm creating this issue so we can track this.

**Comment by DonJayamanne** [comment]
Shouldn't be, this was working until some recent changes (#296).
I know the work is not complete (that PR was primarily a refactor of the existing code)

**Comment by MikhailArkhipov** [comment]
@ericsnowcurrently - does Atatch/Detach merge fix this?

**Comment by ericsnowcurrently** [comment]
This should be fixed by #381.  @DonJayamanne, could you verify?

**Comment by DonJayamanne** [comment]
I can confirm this issue has been resolved.
